openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java

```diff
@@ -0,0 +1,78 @@
+package org.openmetadata.service.apps.bundles.insights.search.opensearch;
+
+import java.io.IOException;
+import org.openmetadata.service.apps.bundles.insights.search.DataInsightsSearchInterface;
+import os.org.opensearch.client.Request;
+import os.org.opensearch.client.Response;
+import os.org.opensearch.client.ResponseException;
+import os.org.opensearch.client.RestClient;
+
+public class OpenSearchDataInsightsClient implements DataInsightsSearchInterface {
+  private final RestClient client;
+
+  public OpenSearchDataInsightsClient(RestClient client) {
+    this.client = client;
+  }
+
+  private Response performRequest(String method, String path) throws IOException {
+    Request request = new Request(method, path);
+    return client.performRequest(request);
+  }
+
+  private Response performRequest(String method, String path, String payload) throws IOException {
+    Request request = new Request(method, path);
+    request.setJsonEntity(payload);
+
+    return client.performRequest(request);
+  }
+
+  @Override
+  public void createLifecyclePolicy(String name, String policy) throws IOException {
+    try {
+      performRequest("PUT", String.format("_plugins/_ism/policies/%s", name), policy);
```

Comment:
nit - should we have an Enum of possible supported requests?

Reply:
What's you ridea? Not sure I followed it

Reply:
We have a lot of "PUT" strings in a bunch of places. We could have `HTTPMethod.PUT` instead, and same for DELETE< etc.